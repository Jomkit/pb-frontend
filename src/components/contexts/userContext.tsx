import { createContext } from "react";
import { IUser } from "../../types";

const userContext = createContext({} as IUser);

export default userContext;