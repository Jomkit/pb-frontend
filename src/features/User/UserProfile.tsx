import { useContext } from 'react'
import Frame from '../../components/Frame'
import userContext from '../../components/contexts/userContext'
import SurveyReport from '../survey/SurveyReport';
import Statistics from './Statistics';

const UserProfile = () => {
  const user = useContext(userContext);
  
  return (
    <Frame bgColor='bg-green-400' title='User Profile'>
      <div className='w-full lg:w-2/3 m-auto p-4 h-fit bg-white rounded-lg'>

        <div className='w-full lg:w-2/3 m-auto p-4 simple-border'>
          <h2 className='text-bold'>{user.username}</h2>
          <div className='text-start'>
            <p>First Name: {user.firstName}</p>
            <p>Last Name: {user.lastName}</p>
            <p>Email Address : {user.email}</p>
            
          </div>
        </div>

        {user.id && <SurveyReport userId={user.id!} />}

        <Statistics userId={user.id!} />
      </div>
    </Frame>
  )
}

export default UserProfile