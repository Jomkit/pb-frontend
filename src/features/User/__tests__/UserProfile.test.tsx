import { render } from "@testing-library/react"
import UserProfile from "../UserProfile"
import userContext from "../../../components/contexts/userContext";

beforeEach(() => {
    vi.mock("../../../api");
})

afterEach(() => {
    vi.restoreAllMocks();
})

const testUser = {
    id: 1,
    username: "testUser",
    firstName: "testFirst",
    lastName: "testLast",
    email: "test@email.com"
}

it("should render without crashing", () => {
    render(
        <userContext.Provider value={testUser}>
            <UserProfile />
        </userContext.Provider>
    );
})

it("should match snapshot", async () => {
    const { asFragment, findByText } = render(
        <userContext.Provider value={testUser}>
            <UserProfile />
        </userContext.Provider>
    );

    const testProject1 = await findByText("testProject 1");
    expect(testProject1).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
})