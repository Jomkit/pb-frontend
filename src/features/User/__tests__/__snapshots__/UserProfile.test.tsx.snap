// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="text-center items-start min-h-screen w-screen p-4 bg-green-400"
  >
    <h1>
      User Profile
    </h1>
    <div
      class="p-10 justify-center"
    >
      <div
        class="w-full lg:w-2/3 m-auto p-4 h-fit bg-white rounded-lg"
      >
        <div
          class="w-full lg:w-2/3 m-auto p-4 simple-border"
        >
          <h2
            class="text-bold"
          >
            testUser
          </h2>
          <div
            class="text-start"
          >
            <p>
              First Name: testFirst
            </p>
            <p>
              Last Name: testLast
            </p>
            <p>
              Email Address : test@email.com
            </p>
          </div>
        </div>
        <div
          class="w-full lg:w-2/3 m-auto mt-10 p-4 simple-border"
        >
          <h2
            class="text-bold"
          >
            Report
          </h2>
          <div
            class="text-start"
          >
            <p>
              Productive time today: 10 minutes, 0 milliseconds
            </p>
            <p>
              Total time last week: 10 minutes, 0 milliseconds
            </p>
            <p>
              Most productive day of the week (by average score): Thursday (1.5)
            </p>
          </div>
        </div>
        <div>
          <div>
            My Statistics
          </div>
          <table
            class="table-auto w-full simple-border"
          >
            <thead
              class="bg-green-200 simple-border"
            >
              <tr>
                <th>
                  Project
                </th>
                <th>
                  Status
                </th>
                <th>
                  Total Time Spent
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  testProject 1
                </td>
                <td>
                  Active
                </td>
                <td>
                  5 minutes
                </td>
              </tr>
              <tr>
                <td>
                  testProject 2
                </td>
                <td>
                  Active
                </td>
                <td>
                  5 minutes
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
